Service/MergedAssetsWarmupUrlsProvider.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace MageOps\NodeWarmer\Service;
+
+class MergedAssetsWarmupUrlsProvider
+{
+    const ALLOWED_VISIBILITIES = [
+        \Magento\Catalog\Model\Product\Visibility::VISIBILITY_IN_CATALOG,
+        \Magento\Catalog\Model\Product\Visibility::VISIBILITY_BOTH,
+    ];
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory
+     */
+    protected $productCollectionFactory;
+
+    /**
+     * @var \Magento\Catalog\Model\ResourceModel\Category\Collection
+     */
+    protected $categoryCollectionFactory;
+
+    /**
+     * @var \Magento\UrlRewrite\Model\UrlFinderInterface
+     */
+    protected $urlFinder;
+
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    protected $url;
+
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productCollectionFactory,
+        \Magento\Catalog\Model\ResourceModel\Category\CollectionFactory $categoryCollectionFactory,
+        \Magento\UrlRewrite\Model\UrlFinderInterface $urlFinder,
+        \Magento\Framework\UrlInterface $url
+    )
+    {
+        $this->storeManager = $storeManager;
+        $this->productCollectionFactory = $productCollectionFactory;
+        $this->categoryCollectionFactory = $categoryCollectionFactory;
+        $this->urlFinder = $urlFinder;
+        $this->url = $url;
+    }
+
+    public function getUrls()
+    {
+        $urls = [];
+
+        foreach ($this->storeManager->getStores() as $store) {
+            $baseUrl = $store->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_WEB);
+
+            $urls[] = $baseUrl;
+            $urls[] = $store->getUrl('customer/account/login');
+            $urls[] = $store->getUrl('customer/account/create');
+            $urls[] = $store->getUrl('customer/account/forgotpassword');
+            $urls[] = $store->getUrl('checkout/cart');
+            $urls[] = $store->getUrl('catalogsearch/result', ['_query' => ['q' => 'test']]);
+            $urls[] = $this->getProductUrl($store);
+            $urls[] = $this->getCategoryUrl($store);
```

Comment:
I'm not sure, but maybe we can add a possibility to add custom URLs here, per project?